### PR TITLE
Add QA deployment verification doc

### PR DESCRIPTION
**Pull request by pdehaan**
r? @clouserw @ckprice

### CODE REVIEW

File: docs/DEPLOYMENT-VERIFICATION.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# Test Pilot Test Plan
+
+## ENVIRONMENTS:
```
Comment (ckprice) [inline docs/DEPLOYMENT-VERIFICATION.md:3]: Do you think it's worth mentioning the `Preferences` drop-down in this section? Like, "this _should_ match with the drop-down, buuuut..."

**Comment by ckprice** [comment]
Thanks for this @pdehaan! Wondering if we should add a link to [the deployment documentation](https://github.com/mozilla/testpilot/blob/master/docs/DEPLOYMENT.md#test-stage) as well? Right now it just links to the Google Doc, but this is way better.

**Comment by ghost** [comment]
love it. r+

File: docs/DEPLOYMENT-VERIFICATION.md (hunk 1 above)
Comment (pdehaan) [inline docs/DEPLOYMENT-VERIFICATION.md:3]: I do, added to the wall of verification steps below, so people remember to confirm after installing the add-on.

**Comment by pdehaan** [comment]
> @ckprice: Wondering if we should add a link to the deployment documentation as well?

Added a note in the DEPLOYMENT.md file below the Google Docs link:

> _Follow the steps in the ["Test Pilot Deployment Verification Test Plan" doc](DEPLOYMENT-VERIFICATION.md) to verify that the site works as expected._

**Pull request merged**